tests/components/insteon/test_init.py

```diff
@@ -0,0 +1,227 @@
+        await hass.async_block_till_done()
+        # pylint: disable=no-member
+        assert insteon.devices.async_save.call_count == 1
+        assert mock_close.called
+
+
+async def test_import_plm(hass: HomeAssistantType):
+    """Test setting up the entry from YAML to a PLM."""
+    config = {}
+    config[DOMAIN] = MOCK_IMPORT_CONFIG_PLM
+
+    with patch.object(
+        insteon, "async_connect", new=mock_successful_connection
+    ), patch.object(insteon, "close_insteon_connection"), patch.object(
+        insteon, "devices", new=MockDevices()
+    ), patch(
+        PATCH_CONNECTION, new=mock_successful_connection
+    ):
+        assert await async_setup_component(
+            hass,
+            insteon.DOMAIN,
+            config,
+        )
+        await hass.async_block_till_done()
+        await asyncio.sleep(0.01)
+    assert hass.config_entries.async_entries(DOMAIN)
+    data = hass.config_entries.async_entries(DOMAIN)[0].data
+    assert data[CONF_DEVICE] == MOCK_IMPORT_CONFIG_PLM[CONF_PORT]
+    assert CONF_PORT not in data
+
+
+async def test_import_hub1(hass: HomeAssistantType):
+    """Test setting up the entry from YAML to a hub v1."""
+    config = {}
+    config[DOMAIN] = MOCK_IMPORT_MINIMUM_HUB_V1
+
+    with patch.object(
+        insteon, "async_connect", new=mock_successful_connection
+    ), patch.object(insteon, "close_insteon_connection"), patch.object(
+        insteon, "devices", new=MockDevices()
+    ), patch(
+        PATCH_CONNECTION, new=mock_successful_connection
+    ):
+        assert await async_setup_component(
+            hass,
+            insteon.DOMAIN,
+            config,
+        )
+        await hass.async_block_till_done()
+        await asyncio.sleep(0.01)
+        assert hass.config_entries.async_entries(DOMAIN)
+    data = hass.config_entries.async_entries(DOMAIN)[0].data
+    assert data[CONF_HOST] == MOCK_IMPORT_FULL_CONFIG_HUB_V1[CONF_HOST]
+    assert data[CONF_PORT] == PORT_HUB_V1
+    assert CONF_USERNAME not in data
+    assert CONF_PASSWORD not in data
+
+
+async def test_import_hub2(hass: HomeAssistantType):
+    """Test setting up the entry from YAML to a hub v2."""
+    config = {}
+    config[DOMAIN] = MOCK_IMPORT_MINIMUM_HUB_V2
+
+    with patch.object(
+        insteon, "async_connect", new=mock_successful_connection
+    ), patch.object(insteon, "close_insteon_connection"), patch.object(
+        insteon, "devices", new=MockDevices()
+    ), patch(
+        PATCH_CONNECTION, new=mock_successful_connection
+    ):
+        assert await async_setup_component(
+            hass,
+            insteon.DOMAIN,
+            config,
+        )
+        await hass.async_block_till_done()
+        await asyncio.sleep(0.01)
+        assert hass.config_entries.async_entries(DOMAIN)
+    data = hass.config_entries.async_entries(DOMAIN)[0].data
+    assert data[CONF_HOST] == MOCK_IMPORT_FULL_CONFIG_HUB_V2[CONF_HOST]
+    assert data[CONF_PORT] == PORT_HUB_V2
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)